### PR TITLE
Don't schedule pods in fargate nodes

### PR DESCRIPTION
**Pull request by limed**
Issue #177 

Description of changes:
This resolves an issue where the aws calico pods tries to get scheduled on a fargate pod, adding affinity to make sure that it will not be scheduled on fargate.

This solution is taken from a similar problem with the `aws-node-termination-handler` chart in commit 86c6e24eab7e22f5edfda7884a827c5119d2c2c4 so if there is a different method of solving this let me know

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: stable/aws-calico/templates/daemon-set.yaml [hunk 1]
```diff
@@ -19,6 +19,15 @@ spec:
...
       priorityClassName: system-node-critical
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                - key: "eks.amazonaws.com/compute-type"
+                  operator: NotIn
+                  values:
+                    - fargate
```
Comment (mogren) [inline stable/aws-calico/templates/daemon-set.yaml:30]: Thanks, looks good. We should probably add checks for OS and Arch as well, since the default Calico image will only work on Linux on amd64. That can be a separate PR.

**Review by mogren — APPROVED** [review]
🚀

**Review by jaypipes — APPROVED** [review]
:+1: awesome, thank you @limed! :)
